### ISSUE TITLE
[Swiglu] question about swiglu 

### ISSUE THREAD

**Issue by mynewstart** [epfLLM/Megatron-LLM#64]
Hi,
I found the following code in arguments.py was deleted in your implementation, can i know why?
```
    if args.swiglu:
        # reduce the dimnesion for MLP since projections happens on
        # two linear layers. this keeps the number of paramters in
        # the same ballpark as the counterpart with 4*h size
        # we keep it a multiple of 64, which means the actual tensor size
        # will be a multiple of 64 / tp_size
        args.ffn_hidden_size = int((4 * args.hidden_size * 2 / 3) / 64) * 64
```

**Comment by mynewstart** [comment]
Hi, thanks for replying! I want to pre-train the llama model, but the ffn_hidden_size in checkpoint file is not same as config.ffn_hidden_size as it adjusted by term of 2 / 3. So I pruned the model weights, but I believe it's not a good choice. In your code, as the ffn_hidden_size won't adjust, so the loading process don't have any problem.

**Comment by kylematoba** [comment]
Sorry, I'm not quite sure what the problem is. It's probably my fault, since I'm not too familiar with this part of the code. But: I don't think I've encountered any problems with checkpoints not loading due to wrong dimensions. Is this a question about our conventions? Do you think we've made some problematic error?

**Comment by mynewstart** [comment]
I apologize for the unclear expression. In fact, my question is that the following code in "Megatron-LM" is included, but it is not present in your codebase. Can you provide a reason for commenting out this code? Is it solely related to training efficiency?
```
    if args.swiglu:
        # reduce the dimnesion for MLP since projections happens on
        # two linear layers. this keeps the number of paramters in
        # the same ballpark as the counterpart with 4*h size
        # we keep it a multiple of 64, which means the actual tensor size
        # will be a multiple of 64 / tp_size
        args.ffn_hidden_size = int((4 * args.hidden_size * 2 / 3) / 64) * 64

```

**Comment by kylematoba** [comment]
Hi, sorry also. I'm probably being needlessly confusing because I don't 100% understand the context.

But I think the short answer is: we dropped the `swiglu` argument because we generalized to different GLU-like activations. Then, strictly speaking, this logic did not work anymore so it probably got dropped, though it could have / should have been generalized also. So: it might be a bug. But I'm kind of surprised that we're able to load checkpoints, etc. if the dimensions mismatch. What would be the consequence of mistakenly omitting this tweak? 

I'm pretty certain it was not for training efficiency.
